lib/python/picongpu/plugins/plot_mpl/energy_waterfall_visualizer.py

```diff
@@ -0,0 +1,169 @@
+        max_iter = max(iteration * 1.39e-16 * 1.e12)
+        self.plt_obj = ax.imshow(np_data, aspect="auto",
+                                 norm=LogNorm(), origin="lower",
+                                 extent=(0, max_iter, 0, max(bins*1.e-3)))
+        self.plt_lin = ax.axvline(self.itera * 1.39e-16 * 1.e12,
+                                  color='#FF6600')
+        self.cbar = plt.colorbar(self.plt_obj, ax=self.ax)
+        self.cbar.set_label(r'Count')
+        ax.set_xlabel('time [ps]')
+        ax.set_ylabel('Energy [MeV]')
+
+    def _update_plt_obj(self):
+        """
+        Implementation of base class function.
+        """
+        counts, bins, iteration = self.data
+        np_data = np.zeros((len(bins), len(iteration)))
+        for index, ts in enumerate(iteration):
+            np_data[:, index] = counts[ts]
+        self.plt_obj.set_data(np_data)
+        self.plt_lin.remove()
+        self.plt_lin = self.ax.axvline(self.itera * 1.39e-16 * 1.e12,
+                                       color='#FF6600')
+        self.plt_obj.autoscale()
+        self.ax.relim()
+        self.ax.autoscale_view(True, True, True)
+        self.cbar.update_normal(self.plt_obj)
+
+    def visualize(self, ax=None, **kwargs):
+        """
+        Creates a semilogy plot on the provided axes object for
+        the data of the given iteration using matpotlib.
+
+        Parameters
+        ----------
+        iteration: int
+            the iteration number for which data will be plotted.
+        ax: matplotlib axes object
+            the part of the figure where this plot will be shown.
+        kwargs: dictionary with further keyword arguments, valid are:
+            species: string
+                short name of the particle species, e.g. 'e' for electrons
+                (defined in ``speciesDefinition.param``)
+            iteration: int
+                number of the iteration
+            species_filter: string
+                name of the particle species filter, default is 'all'
+                (defined in ``particleFilters.param``)
+
+        """
+        self.ax = self._ax_or_gca(ax)
+        self.itera = kwargs.get('iteration')
+        kwargs['iteration'] = None
+        # this already throws error if no species or iteration in kwargs
+        super(Visualizer, self).visualize(ax, **kwargs)
+        species = kwargs.get('species')
+        species_filter = kwargs.get('species_filter', 'all')
+        ax.set_title('Energy Histogram for species ' +
+                     species + ', filter = ' + species_filter)
+
+    def clear_cbar(self):
+        """Clear colorbar if present."""
+        if self.cbar is not None:
+            self.cbar.remove()
+
+
+if __name__ == '__main__':
+
+    def main():
+        import sys
+        import getopt
+
+        def usage():
+            print("usage:")
+            print(
+                "python", sys.argv[0],
+                "-p <path to run_directory> -i <iteration>"
```

Comment:
here the iteration is "only" used for adding a progress line as well.
that's very good for the GUI, but should be optional as a parameter and just be skipped in case a user does not provide it.